### PR TITLE
Fixing Mem-leaks in DataExplorer

### PR DESCRIPTION
**Pull request by rinkk**
this should fix the remaining leaks after PR #1091 on DataExplorer startup.

### CODE REVIEW

File: Applications/DataExplorer/VtkVis/VisualizationWidget.cpp [hunk 1]
```diff
@@ -45,7 +45,7 @@
 #include <QString>
 
 VisualizationWidget::VisualizationWidget( QWidget* parent /*= 0*/ )
-	: QWidget(parent)
+	: _vtkRender(nullptr), _markerWidget(nullptr), QWidget(parent)
```
Comment (endJunction) [inline Applications/DataExplorer/VtkVis/VisualizationWidget.cpp]: warning: field '_markerWidget' will be initialized after base 'QWidget'
The order either in declarations or in the initialization list should be change and be the same.

**Comment by chleh** [comment]
Looks good. :+1:

File: Applications/DataExplorer/VtkAct/VtkCustomInteractorStyle.h [hunk 2]
```diff
@@ -72,13 +72,13 @@ public slots:
...
-	vtkDataObject* Data;
+	vtkDataObject* _data;
 
 	/// @brief The mapper for highlighting the selected cell.
-	vtkDataSetMapper* selectedMapper;
+	vtkDataSetMapper* _selectedMapper;
 
 	/// @brief The actor for highlighting the selected cell.
-	vtkActor* selectedActor;
+	vtkActor* _selectedActor;
```
Comment (TomFischer) [inline Applications/DataExplorer/VtkAct/VtkCustomInteractorStyle.h]: Maybe it is possible to use `vtkSmartPointer` or `std::unique_ptr` for `_data` (line 75), `_selectedMapper` (line 78) or `_selectedActor`?

**Comment by TomFischer** [comment]
One tiny question to answer, then :fast_forward:

File: Applications/DataExplorer/VtkAct/VtkCustomInteractorStyle.h (hunk 2 above)
Comment (rinkk) [inline Applications/DataExplorer/VtkAct/VtkCustomInteractorStyle.h]: I don't think so. Whenever the variables are used, VTK expects "normal" pointers, so I'd have to use ```.get()``` which makes the whole concept debatable in my opinion. On top of that I get error messages from stdlib when the unique_ptr should be deleted but VTK still uses the memory. If it's possible to do it, it's at least not straightforward.

**Pull request merged**